### DISCUSSION TITLE
Per-Display Option for "Gaps between windows"

### DISCUSSION THREAD

**Discussion by williamcpierce** [rxhanson/RectanglePro-Community#489] [Ideas]
I enjoy using the "Gaps between windows" setting on my ultrawide monitor, since I have an excess of screen real estate. 

However, when I switch to using my laptop without a monitor, screen space is at a premium, and I prefer to have no gaps between windows. I have to change this manually every time I disconnect from my monitor. 

Is there any way to specify per-display settings for the "Gaps between windows option"? If not, I'd love to have this feature added.

Thanks for the great app Ryan!

**Comment by rxhanson** [comment]
Sorry for the delayed response. Thanks for the feedback! This would fit in nicely at a time when I add more support for other display specific settings. I'll add it to my list for that feature.

**Reply by pgitkn** [reply]
And taking this suggestion a step further...  "Gaps per app" would be nice too...   My #2 (after RectanglePro) add-on app is DefaultFolderX but the only way I can use it is by adding a 60 pixel gap in Rectangle.  I really only need this gap in Finder but it applies against all apps whether I like it or not.  I tried creating a CUSTOM Size & Position so I could specify dimensions after moving them to my primary display but it still adds the Gap so now I either live w/o DFX or I'm back to OPTION+yellow_button to maximize windows.

![Rectangle CUSTOM](https://github.com/rxhanson/RectanglePro-Community/assets/147437056/e5aae2c4-1cbd-4a93-a884-9d7d5fe0d812)

**Reply by rxhanson** [reply]
Thanks for the additional idea here. I typically don't add features that cater to other specific apps like this, but would reconsider with enough demand, or if it turns out that there are more use cases that I'm not aware of.

**Comment by stujumpsrope** [comment]
How do we overcome this problem of assigning.custom layouts over two monitors ? I have the same setup with a laptop and large LG monitor and love throwing windows around on the main monitor, the LG.  Only thing is when I go to laptop only my custom footprints are off-centre. I have tried a mixture a pixel and percentage but I found out quick throw doesn’t support percentages. All my custom footprints are very specific to using tiny windows on plenty of screen real-estate (mail app 1/8th size centre left; finder window 1/8th centre-right etc). I looked into defining the display D1 & D2 etc, and honestly I don’t know how it works. I’ve tried all the options such as ‘unchanged’ and defining D1 and D2 but no success. What I thought might work is if I define D2 as the target monitor being the LG, when I go laptop only it automatically becomes monitor D2 and my percentage-defined footprints would scale accordingly. But I found the laptop is always D1. Can this be changed in say, system settings? Is there another way to do this?

**Reply by rxhanson** [reply]
I think what you're looking for is defining profiles for each display, so that one shortcut will execute a different size/position depending on the display. If I am understanding correctly, at the moment, there isn't something specifically built into Rectangle Pro to handle this scenario. 

The display numbering is based on the location of the display in System Settings. Dragging the displays around there will adjust the x/y coordinates of the display. D1 would always be the upper left most display.
